### DISCUSSION TITLE
Workbench & Drawing

### DISCUSSION THREAD

**Discussion by geopast** [TerriaJS/terriajs#5152] [Ideas] [answered]
Hi folks,

I'm new to Terria and wondered if the following has been considered or could be for the future:

1. Workbench groups - an ability to group datasets within the workbench
2. Drawing tools - allowing the user to annotate and/or draw on top of existing layers

Thanks,
Neil

**Comment by tephenavies** [comment, marked as answer]
Hi @geopast. 

We have the concept of a CompositeCatalogItem to add/remove multiple datasets from the map. I'm not sure what state it's in currently (either in Terria version 7 or version 8). I think it works to the extent that you can use it to link multiple datasets to be added in 1 click, but you lose a lot of the workbench functionality of both/all linked datasets when using it.

We have thought more about drawing tools. There's been some proposals/thoughts from the community around these drawing tools. I think the most up to date thread about these is in https://github.com/TerriaJS/terriajs/issues/4843. You could ask there if there's any progress, or add your own thoughts

**Reply by geopast** [reply]
Thanks @steve9164,
I've posted some thoughts to #4843 regarding drawing. 
For the "groups" idea, I was thinking it would be useful to be able to put layers in a folder structure - particularly if you have 30+ layers. That way you wouldn't lose the functionality on the individual layers and you could provide a cleaner workbench experience! Some controls on how the folder displays on load could be good - isShown, isOpen etc.
